sdk-core/src/main/java/care/data4life/sdk/network/Environment.kt

```diff
@@ -15,79 +15,71 @@
+import care.data4life.sdk.network.NetworkingContract.Companion.D4L_SP_CERT
+import care.data4life.sdk.network.NetworkingContract.Companion.PLATFORM_D4L
+import care.data4life.sdk.network.NetworkingContract.Companion.PLATFORM_S4H
+import care.data4life.sdk.network.NetworkingContract.Companion.S4H_CERT
 
+enum class Environment : NetworkingContract.Environment {
     LOCAL,
     DEVELOPMENT,
     STAGING,
     SANDBOX,
     PRODUCTION;
 
     override fun getApiBaseURL(platform: String): String {
-        if (D4L.equals(platform, ignoreCase = true)) {
-            return d4lBaseUrl()
-        } else if (S4H.equals(platform, ignoreCase = true)) {
-            return s4hBaseUrl()
+        return when {
+            PLATFORM_D4L.equals(platform, ignoreCase = true) -> d4lBaseUrl()
+            PLATFORM_S4H.equals(platform, ignoreCase = true) -> s4hBaseUrl()
+            else -> throw IllegalArgumentException("No supported platform found for value($platform)")
         }
-        throw IllegalArgumentException("No supported platform found for value($platform)")
     }
 
     private fun d4lBaseUrl(): String {
-        return when (this) {
-            SANDBOX -> "https://api-phdp-sandbox.hpsgc.de"
-            DEVELOPMENT -> "https://api-phdp-dev.hpsgc.de"
-            STAGING -> "https://api-staging.data4life.care"
-            LOCAL -> "https://api.data4life.local"
-            PRODUCTION -> "https://api.data4life.care"
-        }
+        return NetworkingContract.Data4LifeURI.valueOf(this.name).uri
     }
 
     private fun s4hBaseUrl(): String {
-        return when (this) {
-            SANDBOX -> "https://api-sandbox.smart4health.eu"
-            DEVELOPMENT -> "https://api-dev.smart4health.eu"
-            STAGING -> "https://api-staging.smart4health.eu"
-            LOCAL -> "https://api.smart4health.local"
-            PRODUCTION -> "https://api.smart4health.eu"
-        }
+        return NetworkingContract.Smart4HealthURI.valueOf(this.name).uri
     }
 
     override fun getCertificatePin(platform: String): String {
-        if (D4L.equals(platform, ignoreCase = true)) {
-            return d4lCertificatePin()
-        } else if (S4H.equals(platform, ignoreCase = true)) {
-            return sh4CertificatePin()
+        return when {
+            PLATFORM_D4L.equals(platform, ignoreCase = true) -> d4lCertificatePin()
+            PLATFORM_S4H.equals(platform, ignoreCase = true) -> sh4CertificatePin()
+            else -> throw IllegalArgumentException("No supported platform found for value($platform)")
         }
-        throw IllegalArgumentException("No supported platform found for value($platform)")
     }
 
     private fun d4lCertificatePin(): String {
         return when (this) {
-            SANDBOX, DEVELOPMENT, LOCAL -> "sha256/3f81qEv2rjHvcrwof2egbKo5MjjSHaN/4DOl7R+pH0E="
-            STAGING, PRODUCTION -> "sha256/AJvjswWs1n4m1KDmFNnTqBit2RHFvXsrVU3Uhxcoe4Y="
+            SANDBOX, DEVELOPMENT, LOCAL -> D4L_SDL_CERT
```

Comment:
the certificates are bound to a domain, I would reflect that in the name: 
* D4L_SDL_CERT is hpsgc.de
* D4L_SP_CERT is data4life.care
* S4H_CERT is smart4health.eu

Reply:
Good point, so more using the domain instead a generic D4l_SOMETHING_CERT name?

Reply:
done